### ISSUE TITLE
The Wbits=8, Abits=8 degrades serverely on CIFAR10 comparing to the 32bits version

### ISSUE THREAD

**Issue by HaFred** [zzzxxxttt/pytorch_DoReFaNet#12]
I tested the `cifar_train_eval.py` with resenet20, lenet5 configuration. And I discovered that after the weights are quantized if activations are further quantized as well, the accuracy drops severely for both network.
And I also noticed that it was due to the activation quantization in `quant_dorefa.py` did not correctly cover the dynamic range in the activation (refer to [pytorch static quantization](https://pytorch.org/tutorials/advanced/static_quantization_tutorial.html#post-training-static-quantization)).
I will make a PR later on.

**Comment by MasLiang** [comment]
Hi. I have another opinion。
This paper said that 'Here we assume the output of the previous layer has passed through a bounded activation function `h`, which ensures `r ∈ [0, 1]`.
This can also seen in `quant_dorefa.py`. You can see 61st line `torch.clamp(x, 0, 1)` 
So the dynamic range is `[0, 1]`.

**Comment by HaFred** [comment]
> Hi. I have another opinion。
> This paper said that 'Here we assume the output of the previous layer has passed through a bounded activation function `h`, which ensures `r ∈ [0, 1]`.
> This can also seen in `quant_dorefa.py`. You can see 61st line `torch.clamp(x, 0, 1)`
> So the dynamic range is `[0, 1]`.

The point is in the experiment it does not align with such an assumption. You could try to reproduce the accuracy drops issue I have mentioned with the same means.

**Comment by zzzxxxttt** [comment]
Hi @HaFred , thanks for your comment, but currently I don't have GPU resource, so I'm afraid I can't check the experiments, feel free to make a PR if you can verify your idea.

**Comment by HaFred** [comment]
> Hi @HaFred , thanks for your comment, but currently I don't have GPU resource, so I'm afraid I can't check the experiments, feel free to make a PR if you can verify your idea.

No problem. I will make a PR when I am available. Cheers~
